pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/AdminProxyHandler.java

```diff
@@ -260,8 +260,13 @@ protected String rewriteTarget(HttpServletRequest request) {
 
         boolean isFunctionsRestRequest = false;
         String requestUri = request.getRequestURI();
-        if (requestUri.startsWith("/admin/v2/functions")
-            || requestUri.startsWith("/admin/functions")) {
+        // Functions workers handle function, sources, and sinks endpoints
+        if (requestUri.startsWith("/admin/v3/functions")
```

Comment:
a better change might using regex to match "/admin/v*/functions", "/admin/v*/sources" and "/admin/v*/sinks".